### PR TITLE
EVG-2284 add logging for scheduler queue consistency

### PR DESCRIPTION
**Pull request by john-m-liu**

### CODE REVIEW

File: scheduler/scheduler.go [hunk 1]
```diff
@@ -148,6 +148,15 @@ func (s *Scheduler) Schedule(ctx context.Context) error {
 					"span":     time.Since(distroStartTime).String(),
 					"duration": time.Since(distroStartTime),
 				})
+				if len(d.runnableTasksForDistro) != len(res.taskQueueItem) {
+					grip.Alert(message.Fields{
+						"runner":       RunnerName,
+						"distro":       d.distroId,
+						"message":      "inconsistency with scheduler input and output",
+						"input_queue":  d.runnableTasksForDistro,
```
Comment (tychoish) [inline scheduler/scheduler.go]: this could be huge.

File: scheduler/scheduler.go [hunk 2]
```diff
@@ -148,6 +148,15 @@ func (s *Scheduler) Schedule(ctx context.Context) error {
 					"span":     time.Since(distroStartTime).String(),
 					"duration": time.Since(distroStartTime),
 				})
+				if len(d.runnableTasksForDistro) != len(res.taskQueueItem) {
+					grip.Alert(message.Fields{
```
Comment (tychoish) [inline scheduler/scheduler.go]: I'd do: `grip.AlertWhen( len(d.runnableTasksForDistro) != len(res.taskQueueItem), ` 

and omit the if block

File: scheduler/scheduler.go [hunk 3]
```diff
@@ -148,6 +148,15 @@ func (s *Scheduler) Schedule(ctx context.Context) error {
...
+						"runner":       RunnerName,
+						"distro":       d.distroId,
+						"message":      "inconsistency with scheduler input and output",
+						"input_queue":  d.runnableTasksForDistro,
+						"output_queue": res.taskQueueItem,
```
Comment (tychoish) [inline scheduler/scheduler.go]: this could also be huge.

File: scheduler/scheduler.go (hunk 2 above)
Comment (john-m-liu) [reply scheduler/scheduler.go]: I left this in since I'm doing more with it now

File: scheduler/scheduler.go (hunk 1 above)
Comment (john-m-liu) [reply scheduler/scheduler.go]: Changed to only be the difference in task IDs

**Pull request merged**